### PR TITLE
Feature: Implemented a complete search bar feature with clean layout.

### PR DESCRIPTION
**Pull request by SUJALSINGH7473**
This PR addresses and resolves issue #36 

In this PR, I've added a fully functional **search bar** to enhance the user experience and improve data searchability based on batch and the Test_name and it searches the data in the **whole data** instead of just single page. Additionally, I've introduced a static filter button to allow users to refine their results based on specific criteria and I will add its functionality code in the next pr to avoid potential conflict.

I've also made significant improvements to the icons under the "Actions" column, ensuring they are clean and responsive for better usability and a more polished interface. These enhancements aim to streamline the user experience and improve overall navigation within the application.
**Demonstration:**

https://github.com/avantifellows/quiz-creator/assets/100035802/852d2374-5cc2-4268-ae4a-9a2d845ab2c2

**Screenshot for the sample work:**

search feature sample image

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,5 @@
+AF_DB_URL="http://localhost:3001"
```
Comment (suryabulusu) [inline .env:1]: dont put this .env file version in PR. i shared this only for local testing

File: src/components/displayTable/Row.tsx [hunk 2]
```diff
@@ -137,13 +137,23 @@ const TableRow = ({
...
             <Edit
-              className="cursor-pointer float-right"
+              className="cursor-pointer"
               onClick={(e) => {
                 router.push(`/Session?type=edit&sessionId=${id}`);
                 e.stopPropagation();
               }}
             />
           </div>
+          <div title="Delete">
```
Comment (suryabulusu) [inline src/components/displayTable/Row.tsx:147]: we dont want to delete sessions. which is why we did not have it earlier.

File: src/utils/FormInputHandling.ts [hunk 3]
```diff
@@ -2,14 +2,36 @@ import { instance } from "./RootClient";
...
-  const items = hasMore ? data.slice(0, -1) : data;
+  const { data } = await instance.get<DbTypes[]>(`api`);
 
+  //  search in the db for the specific keyword
+  let query = string_query == undefined ? "" : string_query.toLowerCase();
+
+  const filteredData = data.filter((item) => {
+
+    // Accessing nested properties of DbTypes
+    let batch_name = item.meta_data?.batch ? item.meta_data?.batch : "";
```
Comment (suryabulusu) [inline src/utils/FormInputHandling.ts:22]: interaction with backend can be better. we have a `db-service` that allows for better filtering using batch IDs. maybe iterating over all sessions is not a good idea.

https://github.com/avantifellows/db-service/blob/main/docs/INSTALLATION.md

for now dont worry about it

File: .env (hunk 1 above)
Comment (SUJALSINGH7473) [reply .env:1]: ok will remove it

**Pull request closed without merging**